webui/react/src/pages/ExperimentDetails/ExperimentSingleTrialTabs.tsx

```diff
@@ -223,10 +230,20 @@ const ExperimentSingleTrialTabs: React.FC<Props> = ({
         tabBarStyle={{ height: 48, paddingLeft: 16 }}
         onChange={handleTabChange}>
         <TabPane key="overview" tab="Overview">
-          <TrialDetailsOverview experiment={experiment} trial={trialDetails as TrialDetails} />
+          {waitingForTrials ? (
+            <Spinner spinning={true} tip="Waiting for trials..." />
+          ) : wontHaveTrials ? (
+            <WontHaveTrials />
+          ) : (
+            <TrialDetailsOverview experiment={experiment} trial={trialDetails} />
+          )}
```

Comment:
this fixes the issues where experiments that `wontHaveTrials` have infinite loading spinners.